### PR TITLE
[DEV-4931] Agency Profile Page Federal Account List Endpoint

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Creates new endpoint to list the Federal Accounts and Treasury accounts along with the gross outlay amounts and the obligated amounts under a specified agency for a given fiscal year.

**Technical details:**
Uses treasury account table, federal account table, and the financial accounts by program activity object class table. Allows for filtering on the the federal account title and / or the treasury account title. Additionally supports pagination & sorting on the titles, obligated amounts, and the gross outlay amounts

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend
4. [N/A] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [N/A] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-4931](https://federal-spending-transparency.atlassian.net/browse/DEV-4931):
    - [x] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No matview changes, no ops tickets needed
```

### CODE REVIEW

File: usaspending_api/agency/v2/views/federal_account_list.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+from typing import Any, List
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, ListMixin
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.generic_helper import get_simple_pagination_metadata
+from usaspending_api.financial_activities.models import FinancialAccountsByProgramActivityObjectClass
+
+
+class FederalAccountList(ListMixin, AgencyBase):
+    """
+    Obtain the list of program activity categories for a specific agency in a
```
Comment (kbard) [inline usaspending_api/agency/v2/views/federal_account_list.py:13]: Program activities?  Should this be federal accounts?

Comment (emily-brents) [reply usaspending_api/agency/v2/views/federal_account_list.py:13]: yea probably

File: usaspending_api/agency/v2/views/federal_account_list.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                "treasury_account__tas_rendering_label",
+                "treasury_account__account_title",
+                "treasury_account__federal_account__account_title",
+                "treasury_account__federal_account__federal_account_code",
+            )
+            .annotate(
+                obligated_amount=Sum("obligations_incurred_by_program_object_class_cpe"),
+                gross_outlay_amount=Sum("gross_outlay_amount_by_program_object_class_cpe"),
+            )
+            .distinct()
```
Comment (kbard) [inline usaspending_api/agency/v2/views/federal_account_list.py:101]: Don't need a distinct here since the annotations will cause a group by.

Comment (emily-brents) [reply usaspending_api/agency/v2/views/federal_account_list.py]: true

File: usaspending_api/api_contracts/contracts/v2/agency/toptier_code/federal_account.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+## TreasuryAccount (object
++ `name` (required, string)
++ `code` (required, string)
++ `obligated_amount` (required, number)
++ `gross_outlay_amount` (required, number)
+
+## PageMetadata (object)
++ `page` (required, number)
++ `next` (optional, number)
```
Comment (kbard) [inline usaspending_api/api_contracts/contracts/v2/agency/toptier_code/federal_account.md:109]: `next` and `previous` nullable

File: usaspending_api/agency/v2/views/federal_account_list.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        results = self.format_results(self.get_federal_account_list())
+        results = results
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "fiscal_year": self.fiscal_year,
+                "page_metadata": page_metadata,
+                "results": results[self.pagination.lower_limit : self.pagination.upper_limit][: self.pagination.limit],
```
Comment (tony-sappe) [inline usaspending_api/agency/v2/views/federal_account_list.py:62]:
```suggestion
                "results": results[self.pagination.lower_limit : self.pagination.upper_limit]
```

**Pull request merged**